src/codegate/extract_snippets/body_extractor.py

```diff
@@ -0,0 +1,104 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from codegate.extract_snippets.message_extractor import (
+    AiderCodeSnippetExtractor,
+    ClineCodeSnippetExtractor,
+    CodeSnippetExtractor,
+    DefaultCodeSnippetExtractor,
+    OpenInterpreterCodeSnippetExtractor,
+)
+
+
+class BodyCodeSnippetExtractor(ABC):
+
+    def __init__(self):
+        # Initialize the extractor in parent class. The child classes will set the extractor.
+        self._snippet_extractor: CodeSnippetExtractor = None
```

Comment:
Nit, to satisfy pydantic we'd probably want something like:
```
 self._snippet_extractor: CodeSnippetExtractor | None = None
```
or:
```
Union[CodeSnippetExtractor, None]
```

and then check for the extractor to be set in `_extract_from_user_messages`.